components/brave_wallet/browser/zcash/zcash_orchard_sync_state.cc

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/zcash/zcash_orchard_sync_state.h"
+
+#include <utility>
+
+#include "base/check_is_test.h"
+#include "base/containers/extend.h"
+#include "brave/components/brave_wallet/browser/zcash/orchard_shard_tree_delegate_impl.h"
+#include "brave/components/brave_wallet/common/zcash_utils.h"
+
+namespace brave_wallet {
+
+ZCashOrchardSyncState::ZCashOrchardSyncState(base::FilePath path_to_database) {
+  storage_ = base::MakeRefCounted<ZCashOrchardStorage>(path_to_database);
+}
+
+ZCashOrchardSyncState::~ZCashOrchardSyncState() {}
+
+// static
+void ZCashOrchardSyncState::OverrideShardTreeManagerForTesting(
+    const mojom::AccountIdPtr& account_id,
+    std::unique_ptr<OrchardShardTreeManager> manager) {
```

Comment:
`manager` isn't being used here so not clear what this method is used for

Reply:
Removed